react/src/components/SearchComponent.jsx

```diff
@@ -1,13 +1,37 @@
-import React from 'react';
+import React, { useState } from 'react';
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
 import { faMagnifyingGlass } from "@fortawesome/free-solid-svg-icons";
 
-const SearchComponent = ({onChange, onClick}) => {
+const SearchComponent = ({ performSearch }) => {
+    // Create a state variable that stores the search term.
+    const [searchTerm, setSearchTerm] = useState("");
+    const [errorMessage, setErrorMessage] = useState("");
+
+    const performSearchInternal = (e) => {
+        e.preventDefault();
+
+        if (searchTerm.length === 0)
```

Comment:
```suggestion
        if (searchTerm.trim().length === 0)
```
acopera si cazul in care search term-ul contine numai whitespaces

Reply:
Am vazut dupa aceea ca faci trim pe onChangeInput.

Reply:
In event-ul `onClick` mereu se pune in state-ul `searchTerm` termenul deja la care a fost aplicat `trim()`. Nu cred ca ar avea sens sa-i facem `trim()` de fiecare data